src/components/LanguagePicker.tsx

```diff
@@ -0,0 +1,456 @@
+import { useEffect, useRef, useState } from "react"
+import { useRouter } from "next/router"
+import { useTranslation } from "next-i18next"
+import {
+  Box,
+  Flex,
+  forwardRef,
+  Input,
+  Menu,
+  MenuDivider,
+  MenuItem,
+  type MenuItemProps,
+  MenuList,
+  type MenuListProps,
+  type MenuProps,
+  Progress as ChakraProgress,
+  ProgressProps,
+  Text,
+} from "@chakra-ui/react"
+
+import type { Lang, LocaleDisplayInfo } from "@/lib/types"
+
+import { BaseLink, type LinkProps } from "@/components/Link"
+
+import progressData from "@/data/translationProgress.json"
+
+import { DEFAULT_LOCALE } from "@/lib/constants"
+
+import { Button } from "./Buttons"
+
+import i18nConfig from "@/../i18n.config.json"
```

Comment:
The reason for this was the need for the `crowdinCode` which is different from our internal `locale` code. The `locales` object only returns a list of our supported locale codes, nothing more.

The other reason for accessing this was to get access to our full list of custom "display names"... both in English and in the language of that locale.

I'll take a look for another possible way to do this.

Reply:
Switched this to use `languages` exported from `@/lib/utils/translations` instead which is a TypeScript object generated from `i18n.config.json` (had missed this in first pass).

This works much better as it has the same info we need here, and enables getting the details of a language by using our locale code as the key, instead of iterating over the list looking for the match. 👍